File: app/src/main/java/com/dev/moviedb/model/dto/GenreDto.java
package com.dev.moviedb.model.dto;


import android.os.Parcel;
import android.os.Parcelable;


public class GenreDto implements Parcelable {
    private Integer id;
    private String name;


    /**
     * @return The id
     */
    public Integer getId() {
        return id;
    }

    /**
     * @return The name
     */
    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return this.name;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(this.id);
        dest.writeString(this.name);
    }

    public GenreDto() {
    }

    protected GenreDto(Parcel in) {
        this.id = (Integer) in.readValue(Integer.class.getClassLoader());
        this.name = in.readString();
    }

    public static final Parcelable.Creator<GenreDto> CREATOR = new Parcelable.Creator<GenreDto>() {
        public GenreDto createFromParcel(Parcel source) {
            return new GenreDto(source);
        }

        public GenreDto[] newArray(int size) {
            return new GenreDto[size];
        }
    };
}

File: app/src/main/java/com/dev/moviedb/mvvm/modelstate/movies/dto/GenreDTO.kt
package com.dev.moviedb.mvvm.modelstate.movies.dto

import android.os.Parcel
import android.os.Parcelable

/**
 *
 * Yamda 1.0.0.
 */
data class GenreDTO(val id: Int, val name: String) : Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readInt(),
            parcel.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(id)
        parcel.writeString(name)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<GenreDTO> {
        override fun createFromParcel(parcel: Parcel): GenreDTO {
            return GenreDTO(parcel)
        }

        override fun newArray(size: Int): Array<GenreDTO?> {
            return arrayOfNulls(size)
        }
    }
}